pkg/tsdb/azuremonitor/metrics/url-builder_test.go

```diff
@@ -65,8 +65,30 @@ func TestURLBuilder(t *testing.T) {
 				ResourceName:        "rn1/rn2/rn3",
 			}
 
-			url := ub.Build()
+			url := ub.BuildMetricsURL()
 			require.Equal(t, url, "default-sub/resourceGroups/rg/providers/Microsoft.NetApp/netAppAccounts/rn1/capacityPools/rn2/volumes/rn3/providers/microsoft.insights/metrics")
 		})
+
+		t.Run("when only resource uri is provided", func(t *testing.T) {
```

Comment:
I'm not sure if this is a golang thing but it seems weird that we don't say what will happen when the only the resource uri is provided right? Idk maybe a good idea to also get a review from a golang expert 😅